ethcore/src/miner/miner.rs

```diff
@@ -649,7 +663,12 @@ impl MinerService for Miner {
 				// Client should send message after commit to db and inserting to chain.
 				.expect("Expected in-chain blocks.");
 			let block = BlockView::new(&block);
-			block.transactions()
+			let txs = block.transactions();
+			// populate sender
+			for tx in &txs {
+				let _sender = tx.sender();
+			}
```

Comment:
Ah. I moved the logic up to "fetch_transactions" so it happens in both of the below loops. Otherwise, still happening outside the lock in parallel. Doubtful it makes too big of a difference, but I don't have data, so leave it to be safe.